test/parallel/test-repl-eval.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const repl = require('repl');
+
+{
+  let evalCalledWithExpectedArgs = false;
+
+  const options = {
+    eval: common.mustCall((cmd, context) => {
+      // Assertions here will not cause the test to exit with an error code
+      // so set a boolean that is checked in process.on('exit',...) instead.
+      evalCalledWithExpectedArgs = (cmd === 'foo\n' && context.foo === 'bar');
+    })
+  };
+
+  const r = repl.start(options);
+  r.context = {foo: 'bar'};
```

Comment:
It [doesn't appear to be](https://github.com/nodejs/node/blob/f167207d0e4bce0161450f9e32e3e28d2a7a74ec/lib/repl.js#L186-L196).